CMakeLists.txt

```diff
@@ -236,6 +248,69 @@ else()
   message(STATUS "PETSc support disabled")
 endif()
 
+# Option: Ginkgo
+if (PRECICE_GinkgoParallelization)
+  find_package(Ginkgo 1.5 REQUIRED)
+  message(STATUS "Found Ginkgo ${GINKGO_PROJECT_VERSION} at ${GINKGO_INSTALL_PREFIX}")
+
+  # We aim for the same configuration as Ginkgo itself and check immediately for device compiler
+  # such that preciceCore inherits all relevant flags directly
+  # 1. Check if Ginkgo was built with CUDA and if CUDA is available on the system
+  if(${GINKGO_BUILD_CUDA})
+    # Necessary for the unified kernels and the qr decomposition
+    include(CheckLanguage)
+    check_language(CUDA)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+    set(CMAKE_CUDA_FLAGS "${GINKGO_CUDA_ARCH_FLAGS} --expt-relaxed-constexpr")
+
+    # Similar to Ginkgo, we don't let CMake deduce the architecture, but inherit the Ginkgo flags
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.18)
+      set(CMAKE_CUDA_ARCHITECTURES OFF)
+    endif()
+    add_compile_definitions("PRECICE_WITH_CUDA")
+    message(STATUS "Enabled CUDA support for preCICE via the Ginkgo library.")
+  endif()
+
+  # 2. Check if Ginkgo was built with HIP and if HIP is available on the system
+  if(${GINKGO_BUILD_HIP})
+    if(NOT DEFINED HIP_PATH)
```

Comment:
It may be a good idea to be forwards compatible with CMake 3.21, which adds HIP as a language similar to CUDA.